id.js

```diff
@@ -15,10 +15,9 @@ Id.prototype[fl.concat] = function(b) {
 };
 
 // Monoid (value must also be a Monoid)
-Id[fl.empty] = function() {
-    return new Id(this.value[fl.empty] ? this.value[fl.empty]() : this.value.constructor[fl.empty]());
+Id.prototype[fl.empty] = function() {
+    return new Id(typeof this.value[fl.empty] === 'function' ? this.value[fl.empty]() : this.value.constructor[fl.empty]());
```

Comment:
I agree, but I'm not sure the spec allows it:

> A value which has a Monoid must provide an `empty` method on itself or its `constructor` object.

`{'fantasy-land/empty': null, constructor: {'fantasy-land/empty': () => …}}` is valid, strictly speaking. Nowhere do we state that the method must be defined in *both* places, nor do we state that if a Fantasy Land property name is present its value must accord with the spec. I'd love to remove one or both of these sources of ambiguity.